### PR TITLE
Talk claim emails

### PR DESCRIPTION
**Pull request by liam-wiltshire**
Generation and sending of required emails for new world talk claims.

Updated unit tests due to additional dependency (event mapper)

### CODE REVIEW

File: src/controllers/TalksController.php [hunk 1]
```diff
@@ -632,6 +653,21 @@ public function getTalkMapper($db, $request)
         return $this->talk_mapper;
     }
 
+    public function setEventMapper(EventMapper $event_mapper)
```
Comment (heiglandreas) [inline src/controllers/TalksController.php]: Just an idea as I'm seeing a lot of setters and getters sprouting over the next commits: Can't we use Factory-Classes for getting the mappers? Either the default mapper is instantiated or you can call a static setter and then the set mapper will be returned. Basically moving the setMapper/getMapper into its own class.

File: src/models/EventMapper.php [hunk 2]
```diff
@@ -453,6 +453,8 @@ public function transformResults($results, $verbose)
                             . $row['ID'] . '/comments/reported';
                         $list[$key]['reported_talk_comments_uri'] = $base . '/' . $version . '/events/'
                             . $row['ID'] . '/talk_comments/reported';
+                        $list[$key]['pending_claims_uri'] = $base . '/' . $version . '/events/'
```
Comment (heiglandreas) [inline src/models/EventMapper.php]: Is it possible to show this URI only to people that actually can do something with it? So AFAIK that'd be event-admins. I find it nicer to only show people stuff they actually can use. And most people will get a 403 (or 401) when calling that URI.

Comment (railto) [inline src/models/EventMapper.php]: I agree that this would be preferable, let's not show stuff that the majority of people can't access

Comment (liam-wiltshire) [inline src/models/EventMapper.php]: @railto - as per comment on  other PR, I'm pretty sure it's inside the can Edit conditional? (Not at pc to check )

Comment (liam-wiltshire) [inline src/models/EventMapper.php]: Yes it is - see https://github.com/liam-wiltshire/joindin-api/blob/e9839cf7164e0388d8cb413959ddf9803ff399f4/src/models/EventMapper.php#L451

**Pull request merged**